jdbc/src/java/org/apache/hive/jdbc/HiveConnection.java

```diff
@@ -592,6 +638,10 @@ public boolean retryRequest(IOException exception, int executionCount, HttpConte
       }
     });
 
+    if (isBrowserAuthMode()) {
+      httpClientBuilder
```

Comment:
wouldn't httpClientBuilder be null if cookieAuth is disabled? We could use http transport with cookieAuth disabled right?

Reply:
I am not sure if you are reviewing this one commit at a time or by combining all the commits. line 626 on the latest code in this branch creates a httpClientBuilder if the cookieAuth is disabled.

Reply:
This is however a good point. Having browser mode without cookie auth doesn't make sense. Since without cookie auth, each http call will open a browser (althought it will be automatically authenticated). I think for the first version may be we should disallow browser auth if cookie-auth is not enabled on the server side. Thoughts?